web/web/api/v1/sns.py

```diff
@@ -0,0 +1,30 @@
+from flask import Flask, request, Blueprint
+import json
+
+sns_api_bp =  Blueprint('sns_api_bp', __name__)
+
+
+@sns_api_bp.route('/sns', methods = ['POST'])
+def subscriber_information():
+    # AWS sends JSON with text/plain mimetype
+    try:
+        js = json.loads(request.data)
+    except:
+        return 'An error occuered'
+    hdr = request.headers.get('x-amz-sns-message-type')
+    # subscribe to the SNS topic
+    if hdr == 'Notification':
+        parse_real_power(js['Message'])
+    elif hdr == 'SubscriptionConfirmation' and 'SubscribeURL' in js:
+        return js['SubscribeURL']
+    return 'OK'
```

Comment:
This still seems pretty weird to me, a couple of things:
- Do we actually need to return anything in any of the code paths?
- According to SNS docs, to confirm subscription we just need to visit the URL. Does `return`ing actually do this for us or should we issue a request to that URL?
- Also, we need to be explicit about which error we catching. Don't just catch all exceptions.
- If we don't need to return anything, let's just log the errors/success

Reply:
The endpoint errors out without a return at the end, I made changes for the second and third bullet point. The function still seems to require me to return something - I also added the logger.